xrpl/async_support/wallet/wallet_generation.py

```diff
@@ -0,0 +1,119 @@
+
+    Returns:
+        A Wallet on the testnet that contains some amount of XRP.
+
+    Raises:
+        XRPLFaucetException: if an address could not be funded with the faucet.
+        XRPLRequestFailureException: if a request to the ledger fails.
+        requests.exceptions.HTTPError: if the request to the faucet fails.
+
+    .. # noqa: DAR402 exception raised in private method
+    """
+    if "dev" in client.url:  # devnet
+        faucet_url = _DEV_FAUCET_URL
+    elif "altnet" in client.url or "test" in client.url:  # testnet
+        faucet_url = _TEST_FAUCET_URL
+    else:
+        raise XRPLFaucetException(
+            "Cannot fund an account with a client that is not on the testnet or devnet."
+        )
+    if wallet is None:
+        wallet = Wallet.create()
+
+    address = wallet.classic_address
+    # The faucet *can* be flakey... by printing info about this it's easier to
+    # understand if tests are actually failing, or if it was just a faucet failure.
+    if debug:
+        print("Attempting to fund address {}".format(address))
+    # Balance prior to asking for more funds
+    starting_balance = await _check_wallet_balance(address, client)
+
+    # Ask the faucet to send funds to the given address
+    await _request_funding(faucet_url, address)
+    # Wait for the faucet to fund our account or until timeout
+    # Waits one second checks if balance has changed
+    # If balance doesn't change it will attempt again until _TIMEOUT_SECONDS
+    is_funded = False
+    for _ in range(_TIMEOUT_SECONDS):
+        await asyncio.sleep(1)
+        if not is_funded:  # faucet transaction hasn't been validated yet
+            current_balance = await _check_wallet_balance(address, client)
+            # If our current balance has changed, then the account has been funded
+            if current_balance > starting_balance:
+                if debug:
+                    print("Faucet fund successful.")
+                is_funded = True
+        else:  # wallet has been funded, now the ledger needs to know the account exists
+            next_seq_num = await _try_to_get_next_seq(address, client)
+            if next_seq_num is not None:
+                wallet.sequence = next_seq_num
+                return wallet
+
+    raise XRPLFaucetException(
+        "Unable to fund address with faucet after waiting {} seconds".format(
+            _TIMEOUT_SECONDS
+        )
+    )
+
+
+async def _check_wallet_balance(address: str, client: Client) -> int:
+    try:
+        return await get_balance(address, client)
+    except XRPLRequestFailureException as e:
+        if e.error == "actNotFound":  # transaction has not gone through
+            return 0
+        # some other error
+        raise
+
+
+async def _request_funding(url: str, address: str) -> None:
+    async with httpx.AsyncClient() as client:
+        response = await client.post(url=url, json={"destination": address})
```

Comment:
Might wanna use a different variable name other than `client`, I was a bit confused with all the other `client`s in this file